### PR TITLE
[MM-54435] Use annotated logger to capture compliance export logs

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
See https://github.com/mattermost/enterprise/pull/1566

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-54435

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/public/model/compliance.go [hunk 1]
```diff
@@ -123,3 +125,17 @@ func (c *Compliance) IsValid() *AppError {
 
 	return nil
 }
+
+// LoggerFields returns the logger annotations reflecting the given compliance job metadata.
+func (c *Compliance) LoggerFields() []mlog.Field {
```
Comment (hanzei) [inline server/public/model/compliance.go:130]: By adding such a helper function, methods creating a `Compliance` can annotate their logger early.

File: server/public/model/compliance.go [hunk 2]
```diff
@@ -6,6 +6,8 @@ package model
 import (
 	"net/http"
 	"strings"
+
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
```
Comment (isacikgoz) [inline server/public/model/compliance.go:10]: This should also be added to the `go.mod` for public/model package right?

Comment (isacikgoz) [reply server/public/model/compliance.go:10]: Sorry, my bad. Just confused from the shared/mlog. For a moment thought we have a public module for `model` only.

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

**Review by sbishel — APPROVED** [review]
LGTM

File: server/channels/app/compliance.go [hunk 3]
```diff
@@ -46,9 +47,10 @@ func (a *App) SaveComplianceReport(job *model.Compliance) (*model.Compliance, *m
 
 	jCopy := job.DeepCopy()
 	a.Srv().Go(func() {
-		err := a.Compliance().RunComplianceJob(jCopy)
+		c.SetLogger(c.Logger().With(job.LoggerFields()...))
```
Comment (hanzei) [inline server/channels/app/compliance.go:50]: Note to self : Add `Clone()`

**Pull request merged**